### PR TITLE
Environment support

### LINKED ISSUE #65: Configuration with environment variables
It would be nice if we could specify environment variables in the configuration file `importer.yml`

This will enable us to isolate secrets (tokens and passwords), and safely commit the configuration file to version control.

### PR DESCRIPTION
**Pull request by safaci2000**
ChangeLog:
  - Adding support to override values using ENV values
  - Adding related documentation
  - Added Unit test.

Fixes #65

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -41,6 +43,38 @@ func Config() *ConfigStruct {
...
+
+	return mapValue
+}
+
+//applyEnvOverrides a bit of a hack to get around a viper limitation.
+// GetStringMap does not apply env overrides, so we have to traverse it again
+// and reset missing values
+func applyEnvOverrides(contexts map[string]interface{}, mapName string, config *viper.Viper) map[string]interface{} {
+	keys := config.AllKeys()
+	filteredKeys := funk.Filter(keys, func(s string) bool { return strings.Contains(s, "contexts.testing")})
```
Comment (randolph-esnet) [inline config/config.go:67]: This is more of a personal preference, but I'm not a fan of these kinds of libraries that do away with type safety and have unclear function signatures. Especially in this case where you can achieve the same thing in the same number of lines:

for _, key := range config.AllKeys() {
  if !strings.Contains(s, "contexts.testing") {
    continue
  }
 value := config.Get(key)
  ...
}

File: config/config.go [hunk 2]
```diff
@@ -41,6 +43,38 @@ func Config() *ConfigStruct {
 	return configData
 }
 
+//setMapValueEnvOverride recursively iterate over the keys and updates the map value accordingly
+func setMapValueEnvOverride(keys []string, mapValue map[string]interface{}, value interface{}) map[string]interface{} {
+	if len(keys) > 1  {
+		innerMapType := fmt.Sprintf("%T", mapValue[keys[0]])
```
Comment (randolph-esnet) [inline config/config.go:49]: Using a type assertion with the optional 'ok' return value would be more straight forward here: https://go.dev/tour/methods/15

e.g. 
innerMapType, ok := mapValue[keys[0]]
if !ok {
  log.Warn("...")
  return mapValue
}

File: config/config.go [hunk 3]
```diff
@@ -41,6 +43,38 @@ func Config() *ConfigStruct {
 	return configData
 }
 
+//setMapValueEnvOverride recursively iterate over the keys and updates the map value accordingly
+func setMapValueEnvOverride(keys []string, mapValue map[string]interface{}, value interface{}) map[string]interface{} {
```
Comment (randolph-esnet) [inline config/config.go:47]: returning a map[string]interface{} implies you are returning a copy of the original, and the original is unmodified. I suggest removing this return value and accepting that the map will be modified in place.

**Review by randolph-esnet — CHANGES_REQUESTED** [review]

File: config/config.go (hunk 2 above)
Comment (safaci2000) [reply config/config.go:49]: that would just return true if the map contains that key.  It doesn't really say anything about even the data type of the innerMapType.

File: config/config.go (hunk 1 above)
Comment (safaci2000) [reply config/config.go:67]: It creates cleaner code and already using funk in several places.

File: config/config.go (hunk 3 above)
Comment (safaci2000) [reply config/config.go:47]: Done

**Comment by randolph-esnet** [comment]
LGTM

**Pull request merged**